fineract-core/src/main/java/org/apache/fineract/batch/exception/ErrorHandler.java

```diff
@@ -71,24 +88,68 @@ private <T extends RuntimeException> ExceptionMapper<T> findMostSpecificExceptio
         return (ExceptionMapper<T>) defaultExceptionMapper;
     }
 
-    private <T> Set<T> createSet(T[] array) {
-        if (array == null) {
-            return Set.of();
-        } else {
-            return Set.of(array);
-        }
-    }
-
     /**
      * Returns an object of ErrorInfo type containing the information regarding the raised error.
      *
      * @param exception
      * @return ErrorInfo
      */
-    public ErrorInfo handle(final RuntimeException exception) {
+    public ErrorInfo handle(@NotNull RuntimeException exception) {
         ExceptionMapper<RuntimeException> exceptionMapper = findMostSpecificExceptionHandler(exception);
         Response response = exceptionMapper.toResponse(exception);
+        MultivaluedMap<String, Object> headers = response.getHeaders();
+        Set<Header> batchHeaders = headers == null ? null
+                : headers.keySet().stream().map(e -> new Header(e, response.getHeaderString(e))).collect(Collectors.toSet());
         return new ErrorInfo(response.getStatus(), ((FineractExceptionMapper) exceptionMapper).errorCode(),
-                JSON_HELPER.toJson(response.getEntity()));
+                JSON_HELPER.toJson(response.getEntity()), batchHeaders);
+    }
+
+    public RuntimeException getMappable(@NotNull Throwable thr) {
+        return getMappable(thr, null, null, null);
+    }
+
+    public RuntimeException getMappable(@NotNull Throwable t, String msgCode, String defaultMsg, String param,
```

Comment:
The `findMostSpecificExceptionHandler` was introduced to dynamically couple exception handlers to exceptions and move away from the "hardcoded" exception handling. I reckon here it is not coupling which exception to be handled by which exception handler, but rather hardcoding which exception to be translated into a "new" exception. I was wondering rather having this `getMappable` method, it would be better to simply map the proper exception handler for this type of exceptions?

That way we dont have hardcoded logic and easier to work with / overwrite them (if needed).

Reply:
In what situation you have a Throwable exception in Fineract which is not RuntimeException?

Reply:
In what situation? Anywhere where Throwable is caught.

Reply:
i am not following you: `but Throwable is not handled by the mappers, only RuntimeException.`. I am assuming it had something to do with the retry feature for commands (hence this PR is about that...) So i am assuming you had a concrete example when an exception was not handled by any mappers. I am wondering what situation was it and why it cannot be handled by the exception mappers.

Reply:
We discussed and it was not new thing, but put them to a common place. Later it might be further improved.